src/Compiler/CompilerOptions.cpp

```diff
@@ -281,7 +283,7 @@ static llvm::cl::opt<std::string, true> customEnvFlagsOpt("customEnvFlags",
     llvm::cl::desc("Override default option env var OnnxMlirEnvOptionName: "
                    "ONNX_MLIR_FLAGS"),
     llvm::cl::value_desc("option env var"), llvm::cl::location(customEnvFlags),
-    llvm::cl::init("ONNX_MLIR_FLAGS"), llvm::cl::cat(OnnxMlirOptions));
+    llvm::cl::init(OnnxMlirEnvOptionName), llvm::cl::cat(OnnxMlirOptions));
```

Comment:
The initial value for `customEnvFlags` is meaningless because it's not set until `llvm::cl::ParseCommandLineOptions` is called. But we need its initial value in `parseCustomEnvFlagsCommandLineOption` before `llvm::cl::ParseCommandLineOptions` is called. So to make things less confusing, it's probably better to simply do `llvm::cl::init("")` (and add some comments about this).

Reply:
@gongsu832 good suggestion, that is exactly what I discovered empirically, good to have it confirmed.